r/tests/testthat/test-Array.R

```diff
@@ -795,6 +795,26 @@ test_that("Array$create() should have helpful error", {
   expect_error(Array$create(list()), "Requires at least one element to infer")
   expect_error(Array$create(list(lgl, lgl, int)), "Expecting a logical vector")
   expect_error(Array$create(list(char, num, char)), "Expecting a character vector")
+
+  # hint at casting if direct fails and casting looks like it might work
+  expect_error(
+    Array$create(as.double(1:10), type = decimal(4, 2)),
+    "You might want to try casting manually"
+  )
+  # until ARROW-15159 is solved casting from integer to decimal requires a
+  # precision of at least 12
+  # precision < 12 => the attempt to cast errors and the original error message
+  # is shown
+  expect_error(
+    Array$create(1:10, type = decimal(11, 2)),
+    "NotImplemented: Extend"
+  )
```

Comment:
Do we want / need this error message test here? I assume the C++ will be resolved (since that restriction seems quite odd!), and we test below that we still get our "try casting manually" error.

Reply:
This was sort of hitting 2 birds with 1 stone. This is actually testing the `try()` attempt (more specifically its failure and fallback to the first error message). I agree this is dangerous as the error message will no longer appear once (once [ARROW-15159](https://issues.apache.org/jira/browse/ARROW-15159) is addressed.

Reply:
Aaah I see! It might be valuable to test that the try doesn't swallow the original message: Is there a different, more obviously broken, `Array$create` that we could do to test that the original message still shows up? I definitely agree that using the thing we have a jira to fix was / would be confusing

Maybe:

```
> Array$create("one", int32())
Error: Invalid: cannot convert
```

Reply:
What we could do is test the error message from `Array$create()` is the same as the one from `vec_to_Array()`.

Reply:
Oh, yeah I like that

Reply:
Done.